### DISCUSSION TITLE
refactor idea: add a tr_sha_hash type

### DISCUSSION THREAD

**Discussion by ckerr** [transmission/transmission#1922] [Ideas]
If we had something like

```
using tr_sha_hash = std::array<std::byte, 20>;
```

and used that everywhere -- this touch a lot of files, but it would also simplify a lot of the code that deals with hashes. Right now we `memcmp (..., TR_SHA_LENGTH)` everywhere and have a lot of signed char* / unsigned char* / void* ambiguity about handles to these hashes.

**Comment by ile6695** [comment]
It does look like more c++ with std types.
Did you mean `SHA_DIGEST_LENGTH`? I couldn't find `TR_SHA_LENGTH`.

**Reply by ckerr** [reply]
Yes, that's what I meant.

This is mostly  done now with `tr_sha1_digest_t`, but there  are still parts of the codebase using hardcoded arrays w/SHA_DIGEST_LENGTH
